### ISSUE TITLE
[Color Scheme - Camphell]: Incorrect Name is defined for 'Foreground', 'Background' etc.

### ISSUE THREAD

**Issue by Priyanka-Chauhan123** [microsoft/terminal#15985]
### Windows Terminal version

1.19.2481.0

### Windows build number

23529.1000

### Other Software

Test Environment:
OS: Windows 11 Version Dev (OS Build 23529.1000)
App: Windows Terminal Preview

### Steps to reproduce

**Repro Steps: **

1.Open Windows Terminal.
2.Open Settings page using 'Ctr+,'
3. Navigate to the 'Color Scheme' and Select 'Camphell'.
4. Now navigate to 'Foreground', 'Background' etc. control and observe the issue.

**User Experience: **
Screen Reader users will not be able to get the color of the background and foreground, as screen reader is not announcing the color name and only announcing it as foreground and background.

**Observation:**
Incorrect tooltip is also defined for the control.

MAS Reference:
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value

Attachment :
[Incorrect Name is defined for for 'Foreground', 'Background' etc..zip](https://github.com/microsoft/terminal/files/12644742/Incorrect.Name.is.defined.for.for.Foreground.Background.etc.zip)

### Expected Behavior

Correct Name should be defined for 'Foreground', 'Background' etc. Color name should be announced which is defined for foreground and background.

### Actual Behavior

Incorrect Name is defined for 'Foreground', 'Background' etc. Screen reader is just announcing it as "Foreground/Background" and not announcing its color.

**Comment by carlos-zamora** [comment]
https://github.com/microsoft/terminal/issues/15983#issuecomment-1723778110
> [@DHowett]: @zadjii-msft for 15985, I wonder if it would be possible to indicate "Foreground (Ochre)" or "Background (Vermillion)"?

There's two parts to this:
1. converting the color value into readable text:
   - We could store this logic in a converter. I'm hoping there's an easy way to convert from hex to something like "ochre" or "red". Worst case, a long switch-case statement will do the trick (🤢). The easiest approach will be to reveal the hex code and iterate from there.
2. updating the automation property (name):
   - I could be wrong, but I remember having some trouble before when the automation property would not update on an existing UI element. You can set it, but it won't update until it's regenerated. Again, I could be misremembering this, but that's definitely something to keep an eye on.

As for the design, yeah, totally agree that we should say "foreground" and the color value. That way the screen reader user knows that they're modifying the value of "foreground".

**Comment by zadjii-msft** [comment]
> The easiest approach will be to reveal the hex code and iterate from there.

Honestly, let's do that unless it's absolutely trivial to convert an RGB to a _localized_ color name.
